components/dashboard/src/teams/TeamSettings.tsx

```diff
@@ -397,3 +406,100 @@ function OrgDefaultWorkspaceImageModal(props: OrgDefaultWorkspaceImageModalProps
+                    },
+                },
+            );
+        },
+        [updateTeamSettings, props.settings, toast],
+    );
+
+    const noClassesSelected = useMemo(() => {
+        return (props.settings?.allowedWorkspaceClasses.length ?? 0) === 0;
+    }, [props.settings?.allowedWorkspaceClasses]);
+
+    if (isError || !classes) {
+        return <div>Something went wrong</div>;
+    }
+
+    if (isLoading) {
+        return <LoadingState />;
+    }
+
+    return (
+        <div className="space-y-4">
+            <CheckboxListField
+                className="mt-2"
+                error={
+                    <>
+                        {updateTeamSettings.isError && updateTeamSettings.error.message.length > 0 && (
+                            <Alert type="error" closable={true} className="rounded-md">
+                                {updateTeamSettings.error.message}
+                            </Alert>
+                        )}
+                    </>
+                }
+            >
+                {classes.map((wsClass) => (
+                    <CheckboxInputField
+                        key={wsClass.id}
+                        id={wsClass.id}
+                        label={wsClass.displayName}
+                        hint={wsClass.description}
+                        checked={(!isChanged && noClassesSelected) || selectedValue.includes(wsClass.id)}
+                        onChange={(checked) => {
+                            const previousValue =
+                                !isChanged && noClassesSelected ? classes.map((e) => e.id) : selectedValue;
+                            setIsChanged(true);
+                            const newVal = checked
+                                ? [...previousValue, wsClass.id]
+                                : previousValue.filter((e) => e !== wsClass.id);
+                            setValidateError(
+                                newVal.length === 0 ? "Should have one workspace class selected at least." : "",
+                            );
+                            setSelectedValue(newVal);
+                        }}
+                        disabled={props.disabled || updateTeamSettings.isLoading}
+                    />
+                ))}
+            </CheckboxListField>
+
+            <div className="flex gap-2 items-center">
+                {!props.disabled && (
+                    <LoadingButton
+                        disabled={props.disabled || !isChanged || validateError.length > 0}
+                        loading={updateTeamSettings.isLoading}
+                        onClick={() => {
+                            handleUpdateTeamSettings(selectedValue);
+                        }}
+                    >
+                        Save
+                    </LoadingButton>
+                )}
+                {validateError.length > 0 && <span className="text-red-600 dark:text-red-400">{validateError}</span>}
```

Comment:
Can we put this on a line before the button like we do with configs? 🙏 
<img width="554" alt="image" src="https://github.com/gitpod-io/gitpod/assets/29888641/3407bc69-ebd1-403c-856c-a96b11a54142">